### ISSUE TITLE
Retrieve status of Gloo Portal CRs from a Prometheus metrics endpoint

### ISSUE THREAD

**Issue by Jimmmz** [k8sgateway/k8sgateway#7876]
### Version

Portal: 1.2.4
Gloo: 1.11.25-ee

### Is your feature request related to a problem? Please describe.

We had an incident on the weekend where one of our engineers was able to push a network policy without doing the proper testing but ultimately it lead to one of our API Products being taken down in Gloo Portal.
Basically we enforce Network Policies on all namespace and the API Portal controller was blocked, therefore couldn't download the versioned APIDoc required. This resulted in an i/o timeout which took down the entire API.
This is to be expected... However I'd like to monitor for this behaviour so we can catch it. Preferably as part of our Prometheus/Grafana stack. There doesn't appear to be any metrics for failed CRs that I can run a query against in Prometheus.
I can't use HTTP Metrics for this as APIs without a valid APIDoc return a 404 which is a valid response.
Is this doable? It's a rare enough scenario but given its enough to cause downtime I need to cover it.
tldr; I need to run a Prometheus query to determine whether any CRs in gloo portal are in a failed state in order to trigger the necessary alerting.

### Describe the solution you'd like

We require the means of retrieving the status of Gloo Portal CRs from a Prometheus metrics endpoint. This is to monitor the status of Gloo Portal CRs to track any failures that may unfold in everyday usage of Gloo Portal. Currently users are unable to track when Gloo Portal CRs are in a faulty state which has the potential to take down entire APIs without the user knowing.
If an exporter route is chosen, the exporter needs to scrape all Gloo Portal CRs at a configurable interval and expose these via a metrics endpoint. Labels could be used to indicate the status of each CR but this a design decision that Solo would be more fit to decide on...

### Describe alternatives you've considered

Has Solo considered writing an exporter to expose CR statuses? We're exploring writing one to cover these scenarios if theres no available solution
